doc/reference/networking/net_l2.rst

```diff
@@ -20,38 +19,40 @@ object. The L2 layer dictates the API provided by the device driver,
 specific for that device, and optimized for working together.
 
 Currently, there are L2 layers for Ethernet, IEEE 802.15.4 Soft-MAC,
-Bluetooth IPSP, and a dummy one, which is a generic layer example that
-can be used as a template for writing a new one.
+Bluetooth IPSP, CANBUS, OpenThread, WiFi and a dummy one, which is a generic
+layer example that can be used as a template for writing a new one.
 
 L2 layer API
 ************
 
 In order to create an L2 layer, or even a driver for a specific L2
-layer, one needs to understand how the IP layer interacts with it and
-how the L2 layer is supposed to behave. The generic L2 API has 3
-functions:
-
-- recv: All device drivers, once they receive a packet which they put
-  into a :c:type:`struct net_pkt`, will push this buffer to the IP
-  core stack via :c:func:`net_recv_data()`. At this point, the IP core
+layer, one needs to understand how the L3 layer interacts with it and
+how the L2 layer is supposed to behave.
+See also :ref:`network stack architecture <network_stack_architecture>` for
+more details. The generic L2 API has these functions:
+
+- ``recv()``: All device drivers, once they receive a packet which they put
+  into a :c:type:`struct net_pkt`, will push this buffer to the network
+  stack via :c:func:`net_recv_data()`. At this point, the network
   stack does not know what to do with it. Instead, it passes the
-  buffer along to the L2 stack's recv() function for handling. The L2
-  stack does what it needs to do with the packet, for example, parsing
-  the link layer header, or handling link-layer only packets. The
-  recv() function will return NET_DROP in case of an erroneous packet,
+  buffer along to the L2 stack's ``recv()`` function for handling.
+  The L2 stack does what it needs to do with the packet, for example, parsing
+  the link layer header, or handling link-layer only packets. The ``recv()``
+  function will return NET_DROP in case of an erroneous packet,
   NET_OK if the packet was fully consumed by the L2, or NET_CONTINUE
-  if the IP stack should then handle it as an IP packet.
+  if the network stack should then handle it.
+
+- ``send()``: Similar to receive function, the network stack will call this
+  function to actually send a network packet. All relevant link-layer content
+  will be generated and added by this function. The ``send()`` function
+  returns negative error code if there was failure to send the network packet
```

Comment:
This would read better as:
```
The ``send()`` function returns the number of bytes sent, or a negative
error code if there was a failure sending the network packet.
```